src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -368,153 +373,151 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm, bool IsPac
+        /// <param name="occurQuery">Occur type can be Should or Must</param>
+        /// <param name="fuzzyLogicMaxEdits">Max edit lenght for Fuzzy queries</param>
+        /// <param name="termSplit">Indicates if the SearchTerm has been split by empty space or not</param>
+        private void AddQueries(string searchTerm, string field, SearchType searchType, BooleanQuery booleanQuery, Occur occurQuery, int fuzzyLogicMaxEdits, bool termSplit = false)
+        {
+            string querySearchTerm = searchTerm.Replace(" ", string.Empty);
+
+            FuzzyQuery fuzzyQuery;
+            if (searchTerm.Length > LuceneConfig.FuzzySearchMinimalTermLength)
+            {
+                fuzzyQuery = new FuzzyQuery(new Term(field, querySearchTerm), fuzzyLogicMaxEdits);
+                booleanQuery.Add(fuzzyQuery, Occur.SHOULD);
+            }
+
+            if (searchType == SearchType.ByDotCategory && field == nameof(LuceneConfig.NodeFieldsEnum.CategorySplitted))
+            {
+                occurQuery = Occur.MUST;
+            }
+
+            foreach (WildcardType enumVal in Enum.GetValues(typeof(WildcardType)))
+            {
+                var wildcardQuery = CalculateFieldWeight(field, querySearchTerm, enumVal, termSplit);
+                booleanQuery.Add(wildcardQuery, occurQuery);
+            }
+        }
+
+        private WildcardQuery CalculateFieldWeight(string fieldName, string searchTerm, WildcardType wilcardType = WildcardType.None, bool termSplit = false)
         {
             WildcardQuery query;
+            bool isWildcard = true;
+            float boostOffset = 0;
 
-            //In case we are weighting the NameSplitted field then means that is a search based on Category of the type "cat.node" so we will be using the wilcard "category.node*" otherwise will be the normal wildcard
-            var termText = fieldName == nameof(LuceneConfig.NodeFieldsEnum.NameSplitted) ? searchTerm + "*" : "*" + searchTerm + "*";
+            if (fieldName == nameof(LuceneConfig.NodeFieldsEnum.NameSplitted))
+            {
+                wilcardType = WildcardType.PostFix;
+            }
+
+            string termText;
+            //If the WilcardQuery contains regular expression then we will decrease the defined weight
+            switch (wilcardType)
+            {
+                case WildcardType.Prefix:
+                    termText = "*" + searchTerm;
+                    boostOffset = 2;
+                    break;
+                case WildcardType.PostFix:
+                    termText = searchTerm + "*";
+                    boostOffset = 1;
+                    break;
+                case WildcardType.FullCard:
+                    termText = "*" + searchTerm + "*";
+                    boostOffset = 3;
+                    break;
+                default:
+                    isWildcard = false;
+                    termText = searchTerm;
+                    break;
+            }
```

Comment:
So if I am getting this right, the more the boostOffset the less the boost will be, since we are subtracting the offset from the set weight for that field.
That means, `num` > `num*` > `*num` > `*num*` ?

Reply:
yes, you are right. For example if you search for "number" just for the Name field we will be creating the next reg-ex Wildcardqueries (thinking that LuceneConfig.SearchNameWeight = 10 and LuceneConfig.WildcardsSearchNameWeight = 7):

Name:number - 10
Name:number* - 6
Name:*number - 5
Name:*number* - 4